federation-js/src/composition/validate/sdl/uniqueTypeNamesWithFields.ts

```diff
@@ -65,9 +65,9 @@ export function UniqueTypeNamesWithFields(
       // By inspecting the diff, we can warn when field types mismatch.
       // A diff entry will exist when a field exists on one type and not the other, or if there is a type mismatch on the field
       // i.e. { sku: [Int, String!], color: [String] }
-      const { kind, fields } = diffTypeNodes(node, duplicateTypeNode);
+      const { kind, fields, inputValues } = diffTypeNodes(node, duplicateTypeNode);
 
-      const fieldsDiff = Object.entries(fields);
+      const fieldsDiff = Object.entries(fields).concat(Object.entries(inputValues));
```

Comment:
does this make sense @trevor-scheer ?

I think we might need to update `checkFieldUniquenessExcludingValueTypes` as well :)

Reply:
I think it might _work_ but I'm not sure the error reporting will be very accurate if we group these diffs into the `fieldsDiff`. Notice down below (L97) when we iterate over `fieldsDiff`, we might push error messages that relate to a field, but the error code and message don't really hint at input values. It seems appropriate to introduce another error code, like `VALUE_TYPE_INPUT_VALUE_MISMATCH` and have some similar iteration over the `inputValuesDiff` which contributes to the `typesHaveSameShape` evaluation. Does that make sense? It's similar to what you're already doing, but it does split out the iteration over `inputValues` so we can provide a more targeted error message.

Feel free to draw inspiration from the format and wording of existing error messages, or I'd be happy to provide guidance there. There's also some documentation around our error codes which would need a matching update:
https://github.com/apollographql/federation/blob/8b5ebc6d868702dc08dc966554ae4515d2114edd/docs%2Fsource%2Ferrors.md#L89

Reply:
I think you're right that an update to checkFieldUniquenessExcludingValueTypes makes sense as well.

Reply:
not really happy with the code duplication, I was thinking of moving the check to a function, but we'd need to change a bit how we report the error, so might not be worth it :)